File: src/2073-Time-Needed-to-Buy-Tickets/time_needed_to_buy_tickets.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    int timeRequiredToBuy(std::vector<int>& tickets, int k) {
        int time_taken = 0;

        for (int i = 0; i < tickets.size(); ++i) {
            if (i <= k) {
                time_taken += std::min(tickets[i], tickets[k]);
            } else {
                time_taken += std::min(tickets[i], tickets[k] - 1);
            }
        }

        return time_taken;
    }
};

int main() {
    std::vector<int> tickets = {2, 3, 2};
    int k = 2;
    Solution solution;

    std::cout << solution.timeRequiredToBuy(tickets, k) << std::endl;

    return 0;
}

File: src/2073-Time-Needed-to-Buy-Tickets/time_needed_to_buy_tickets.py
from typing import List

class Solution:
    def timeRequiredToBuy(self, tickets: List[int], k: int) -> int:
        time_taken: int = 0

        for i in range(len(tickets)):
            if i <= k:
                time_taken += min(tickets[i], tickets[k])
            else:
                time_taken += min(tickets[i], tickets[k] - 1)
        
        return time_taken
            
    
if __name__ == "__main__":
    tickets = [2,3,2]
    k = 2
    solution = Solution()
    
    print(solution.timeRequiredToBuy(tickets, k))